sc-tools/sc-server/src/sc-server-impl/sc-memory-json/sc-memory-json-action/sc_memory_json_actions_handler.cpp

```diff
@@ -88,43 +41,9 @@ ScMemoryJsonPayload ScMemoryJsonActionsHandler::HandleRequestPayload(
     return responsePayload;
   }
 
-  try
-  {
-    auto * action = it->second;
-    responsePayload = action->Complete(m_context, requestPayload);
-
-    result = SC_TRUE;
-  }
-  catch (ScServerException const & e)
-  {
-    responsePayload = e.m_msg;
-  }
-  catch (utils::ScException const & e)
-  {
-    responsePayload = e.Description();
-  }
-  catch (std::exception const & e)
-  {
-    responsePayload = e.what();
-  }
-  catch (...)
-  {
-    responsePayload = "Undefined error occurred";
-  }
-
-  if (result == SC_FALSE)
-  {
-    m_server->LogError(ScServerLogErrors::rerror, responsePayload);
-  }
+  auto * action = it->second;
+  responsePayload = action->Complete(m_context, requestPayload);
 
+  status = SC_TRUE;
```

Comment:
Wouldn't this override the status, even if it was set to `SC_FALSE` when the processing finishes? I am just not sure whether the idea of `status` field is to indicate that the request was processed or that it was processed **successfully** . In the case of the latter, wouldn't it make all our requests successful, even if they aren't?

Reply:
- if status is `SC_TRUE`, then command has been finished **successfully;**
- if status is `SC_FALSE` and response payload contains no error descriptions, but are some result when command finishing without success, then command has been finished **unsuccessfully;**
- if status is `SC_FALSE` and response payload contains error description, then command has been finished **with error**.
Class of command that finished with error is a subset of a class of command that finished unsuccessfully.

Reply:
Since commands usually contain subcommands, in fact each subcommand must have its own status. I decided that it would be better if in clients we refuse to use the status of the entire command and start use something else. For example, I propose to introduce an error field, in which there will be correspondences between the position of the command in the message text and the error. So it turns out that if this field is empty, then there are no errors, all subcommands were processed successfully, that is, the command itself was processed successfully. If the error field is not empty and is an object of pairs, then an exception can be thrown in it by the position of the message text subcommand when this subcommand is processed on the client side. Also, the error field may simply contain a description of the error. This may be the case when the command text is incorrect, for example, or something similar, that is, this error did not occur as a result of processing the command and its subcommands.